<html><body><pre>
yarn run v1.15.2
$ eslint .

/home/jenkins/workspace/_org_ui-plugin-create-item_PR-27/project/CreateItem/CreateItemWrapper.js
  18:5  error  'state' is not defined  no-undef
  20:6  error  Missing semicolon       semi

✖ 2 problems (2 errors, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
</pre><body></html>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Change history for ui-plugin-create-item
 
+## [1.5.0](https://github.com/folio-org/ui-plugin-create-item/tree/v1.5.0) (in progress)
+[Full Changelog](https://github.com/folio-org/ui-plugin-create-item/compare/v1.4.1...v1.5.0)
+
+### Stories
+* [UIREC-32](https://issues.folio.org/browse/UIREC-32) Add piece for a title
+
 ## [1.4.1](https://github.com/folio-org/ui-plugin-create-item/tree/v1.4.1) (2019-12-30)
 [Full Changelog](https://github.com/folio-org/ui-plugin-create-item/compare/v1.4.0...v1.4.1)
 

--- a/CreateItem/CreateItemModal.js
+++ b/CreateItem/CreateItemModal.js
@@ -4,14 +4,12 @@ import { get, keyBy, some } from 'lodash';
 import { FormattedMessage } from 'react-intl';
 
 import ItemForm from '@folio/inventory/src/edit/items/ItemForm';
-import {
-  Modal,
-  omitProps,
-} from '@folio/stripes/components';
+import { stripesConnect } from '@folio/stripes/core';
+import { Modal } from '@folio/stripes/components';
 
 import css from './CreateItemModal.css';
 
-export default class CreateItemModal extends React.Component {
+class CreateItemModal extends React.Component {
   static manifest = {
     identifierTypes: {
       type: 'okapi',
@@ -154,19 +152,16 @@ export default class CreateItemModal extends React.Component {
     }).isRequired,
     closeCB: PropTypes.func.isRequired,
     locationId: PropTypes.string.isRequired,
+    // instanceId prop is used in manifest
+    // eslint-disable-next-line react/no-unused-prop-types
     instanceId: PropTypes.string.isRequired,
     addItem: PropTypes.func.isRequired,
     mutator: PropTypes.object,
-    onCloseModal: PropTypes.func,
-    openWhen: PropTypes.bool,
-    dataKey: PropTypes.string.isRequired,
     resources: PropTypes.object,
   }
 
   constructor(props) {
     super(props);
-
-    this.connectedApp = props.stripes.connect(ItemForm, { dataKey: props.dataKey });
 
     this.state = {
       error: null,
@@ -192,7 +187,7 @@ export default class CreateItemModal extends React.Component {
   }
 
   render() {
-    const { resources } = this.props;
+    const { resources, stripes } = this.props;
     const { error } = this.state;
 
     const referenceTables = {
@@ -234,7 +229,6 @@ export default class CreateItemModal extends React.Component {
       permanentLocationId: this.props.locationId,
     };
     const instance = get(resources, 'instance.records.0', {});
-    const props = omitProps(this.props, ['resources', 'mutator', 'closeCB', 'locationId', 'instanceId']);
     const initialValues = {
       status: { name: 'On order' },
       holdingsRecordId: holdingsRecord.id,
@@ -256,8 +250,9 @@ export default class CreateItemModal extends React.Component {
               <FormattedMessage id={`ui-plugin-create-item.error.${errorKey}`} />
             </div>
           )}
-          <this.connectedApp
-            {...props}
+          <ItemForm
+            okapi={stripes.okapi}
+            stripes={stripes}
             onSubmit={this.onSubmit}
             initialValues={initialValues}
             referenceTables={referenceTables}
@@ -270,3 +265,5 @@ export default class CreateItemModal extends React.Component {
     );
   }
 }
+
+export default stripesConnect(CreateItemModal);

--- a/CreateItem/CreateItemWrapper.js
+++ b/CreateItem/CreateItemWrapper.js
@@ -1,78 +1,74 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import { Button, Icon, omitProps } from '@folio/stripes/components';
 import className from 'classnames';
+
+import {
+  Button,
+  Icon,
+} from '@folio/stripes/components';
 
 import css from './CreateItemWrapper.css';
 import CreateItemModal from './CreateItemModal';
 
-export default class CreateItemWrapper extends React.Component {
-  constructor(props) {
-    super(props);
+const CreateItemWrapper = ({
+  addItem,
+  disabled,
+  instanceId,
+  locationId,
+  marginBottom0,
+  marginTop0,
+  searchButtonStyle,
+  searchLabel,
+}) => {
+  const [isModalOpened, setIsModalOpened] = useState(false);
 
-    this.connectedCreateItemModal = props.stripes.connect(CreateItemModal, { dataKey: props.dataKey });
-  }
+  const toggleModal = () => setIsModalOpened(!isModalOpened);
 
-  state = {
-    openModal: false,
-  }
-
-  getStyle() {
-    const { marginBottom0, marginTop0 } = this.props;
-
+  const getStyle = () => {
     return className(
       css.searchControl,
       { [css.marginBottom0]: marginBottom0 },
       { [css.marginTop0]: marginTop0 },
     );
-  }
+  };
 
-  openModal = () => this.setState({
-    openModal: true,
-  });
-
-  closeModal = () => this.setState({
-    openModal: false,
-  });
-
-  render() {
-    const { disabled, searchButtonStyle, searchLabel } = this.props;
-    const props = omitProps(this.props, ['disabled', 'searchButtonStyle', 'searchLabel', 'marginBottom0', 'marginTop0']);
-
-    return (
-      <div className={this.getStyle()}>
-        <Button
-          data-test-add-item
-          disabled={disabled}
-          key="searchButton"
-          buttonStyle={searchButtonStyle}
-          onClick={this.openModal}
-        >
-          {searchLabel || <Icon icon="search" color="#fff" />}
-        </Button>
-        {this.state.openModal && (
-          <this.connectedCreateItemModal
-            closeCB={this.closeModal}
-            {...props}
-          />
-        )}
-      </div>
-    );
-  }
-}
+  return (
+    <div className={getStyle}>
+      <Button
+        data-test-add-item
+        disabled={disabled}
+        key="searchButton"
+        buttonStyle={searchButtonStyle}
+        onClick={toggleModal}
+      >
+        {searchLabel || <Icon icon="search" color="#fff" />}
+      </Button>
+      {isModalOpened && (
+        <CreateItemModal
+          closeCB={toggleModal}
+          addItem={addItem}
+          locationId={locationId}
+          instanceId={instanceId}
+        />
+      )}
+    </div>
+  );
+};
 
 CreateItemWrapper.defaultProps = {
-  dataKey: 'createItem',
   disabled: false,
   searchButtonStyle: 'primary noRightRadius',
 };
 
 CreateItemWrapper.propTypes = {
-  dataKey: PropTypes.string,
+  addItem: PropTypes.func.isRequired,
+  instanceId: PropTypes.string.isRequired,
+  locationId: PropTypes.string.isRequired,
   disabled: PropTypes.bool,
   searchLabel: PropTypes.node,
   searchButtonStyle: PropTypes.string,
   marginBottom0: PropTypes.bool,
   marginTop0: PropTypes.bool,
-  stripes: PropTypes.object,
 };
+
+export default CreateItemWrapper;

--- a/README.md
+++ b/README.md
@@ -21,10 +21,8 @@ of the Module Developer's Guide.
 | `instanceId` | string | id of instance record from Inventory in scope of what user want to Create a new item record | Yes |
 | `locationId` | string | id of location record from Inventory, which is used to provide a corresponding holding for new item | Yes |
 | `disabled` | boolean | Flag to control `disable` property of plugin's button, since it's rendered inside the plugin | No |
-| `stripes` | object | stripes-core glue object | Yes |
 | `searchButtonStyle` | string | optional styling of plugin's button | No |
 | `searchLabel` | React.node | optional jsx for plugin's button label | No |
-| `dataKey` | string | dataKey passed to stripes/connect when creating the connected component. | No |
 
 This is a [Stripes](https://github.com/folio-org/stripes-core/) UI module for creating item using code of ui-inventory.
 

